gravitee-gateway-handlers/gravitee-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/context/ApiTemplateVariableProvider.java

```diff
@@ -43,9 +59,29 @@ public void afterPropertiesSet() {
     public void provide(TemplateContext templateContext) {
         // Keep this variable for backward compatibility
         if (api.getProperties() != null) {
-            templateContext.setVariable("properties", api.getProperties().getValues());
+            templateContext.setVariable("properties", decryptProperties(api.getProperties()));
         }
 
         templateContext.setVariable("api", apiProperties);
     }
+
+    private Map<String, String> decryptProperties(Properties properties) {
+        return properties
+            .getProperties()
+            .stream()
+            .map(this::decryptProperty)
+            .collect(toMap(Property::getKey, Property::getValue, (existing, replacement) -> existing));
+    }
+
+    private Property decryptProperty(Property property) {
+        if (property.isEncrypted()) {
+            try {
+                property.setValue(dataEncryptor.decrypt(property.getValue()));
```

Comment:
IMHO, the api properties "decryption" will be made on each api call and can be resource consuming. We should find a way to decrypt the api properties only one time (when deploying the api for instance).